sw/airborne/subsystems/imu/imu_mpu9250_spi.c

```diff
@@ -225,19 +236,20 @@ bool_t imu_mpu9250_configure_mag_slave(Mpu9250ConfigSet mpu_set, void *mpu)
   mpu_wait_slave4_ready();
 
   // Set it to continious measuring mode 2
-  mpu_set_and_wait(mpu_set, mpu, MPU9250_REG_I2C_SLV4_ADDR, (AK8963_ADDR >> 1));
+  mpu_set_and_wait(mpu_set, mpu, MPU9250_REG_I2C_SLV4_ADDR, (MPU9250_MAG_ADDR >> 1));
   mpu_set_and_wait(mpu_set, mpu, MPU9250_REG_I2C_SLV4_REG, AK8963_REG_CNTL1);
   mpu_set_and_wait(mpu_set, mpu, MPU9250_REG_I2C_SLV4_DO, AK8963_CNTL1_CM_2);
   mpu_set_and_wait(mpu_set, mpu, MPU9250_REG_I2C_SLV4_CTRL, (1 << 7)); // Slave 4 enable
 
   mpu_wait_slave4_ready();
-
-  mpu_set_and_wait(mpu_set, mpu, MPU9250_REG_I2C_SLV0_ADDR, (AK8963_ADDR >> 1) | MPU9250_SPI_READ);
+    
+  //Config SLV0 for continus Read 
+  mpu_set_and_wait(mpu_set, mpu, MPU9250_REG_I2C_SLV0_ADDR, (MPU9250_MAG_ADDR >> 1) | MPU9250_SPI_READ);
   mpu_set_and_wait(mpu_set, mpu, MPU9250_REG_I2C_SLV0_REG, AK8963_REG_HXL);
   // Put the enable command as last.
   mpu_set_and_wait(mpu_set, mpu, MPU9250_REG_I2C_SLV0_CTRL,
                    (1 << 7) |    // Slave 0 enable
-                   (7 << 0));    // Read 7 bytes (mag x,y,z + status)
+                   (8 << 0));    // Read 8 bytes (mag x,y,z + status, cntl1)
```

Comment:
1. I want use it to check the valid in future. MPU9250 can bind 4 slave IIC devices. As I get from the data sheet,  I afraid it maybe confuse when i add a baro senser.
2. Int16FromBuf is easy to understand in imu_mpu9250_spi.c.